hoover/search/es.py

```diff
@@ -1,17 +1,34 @@
+import json
+from contextlib import contextmanager
 from django.conf import settings
 from elasticsearch import Elasticsearch
 from elasticsearch.client.utils import _make_path
 from elasticsearch.exceptions import NotFoundError, RequestError, ConnectionError
 
-es = Elasticsearch(settings.HOOVER_ELASTICSEARCH_URL)
 DOCTYPE = 'doc'
 
 class SearchError(Exception):
     def __init__(self, reason):
         self.reason = reason
 
+@contextmanager
+def elasticsearch():
```

Comment:
Of course, squashing the commits is a good thing to do, because we have several large arcs of refactoring in there:

- implemented error checking as function wrappers
- turned those into the context manager and used it everythere in `es.py`
- gone from structured query input to `query_string`s
- gone back to structured queries
- gone back to `query_string`s

Advise on how you want to do this.